recipe/meta.yaml

```diff
@@ -32,20 +30,41 @@ outputs:
       files:
         - test.cu
       commands:
+        # Get intial state of $CUDA_HOME
+        - export TEST_CUDA_HOME_INITIAL=$CUDA_HOME
```

Comment:
In hindsight, the environment may be activated with the scripts already at this point... Is there an easy way to grab an environment variable here before the environment is set?

Reply:
Checking the backup environment variable? Looks like it is called `CUDA_HOME_CONDA_NVCC_BACKUP` (unless I misread things). 🙂

Reply:
I moreso want to check that roundtripping activate / deactivate doesn't have unintended side effects and wanted to get the state of things before activating.

Reply:
I don't know of a good way to do that.

Reply:
How could I even save the initial environment state in a test for a package like this? The test executes with the environment activated.

Reply:
Right this is something that would need to be done in the activation script itself. We create [the activation script during the build]( https://github.com/conda-forge/nvcc-feedstock/blob/5e21ee5bf0c8f2bf5829c21521f48c3cff8bc419/recipe/install_nvcc.sh#L13-L37 ). This way we can have access to any environment variables before we modify them. Of course this doesn't eliminate other package activation scripts causing us issues here (if that's your concern). Though I think this latter concern is unlikely to be of issue here.

Reply:
After looking over this more closely, maybe what you want can be accomplished by manually running the activation and deactivation scripts. This would allow you to configure the outside environment as you see fit and then re-run activation to see if it behaved correctly.